test/jdk/jdk/nio/zipfs/TestLocOffsetFromZip64EF.java

```diff
@@ -143,88 +134,87 @@ public void walkZipFileTest() throws IOException {
-        } catch (IOException e) {
-            throw new RuntimeException(
-                    format("Couldn't read process output '%s'", pb.command()), e);
-        }
+        ByteBuffer buffer = ByteBuffer.wrap(bytes).order(ByteOrder.LITTLE_ENDIAN);
 
-        try {
-            p.waitFor();
-        } catch (InterruptedException e) {
-            throw new RuntimeException(
-                    format("Process hasn't finished '%s'", pb.command()), e);
-        }
-        return new Result(p.exitValue(), output);
-    }
+        // Local header name length
+        short nlenLoc = buffer.getShort(ZipFile.LOCNAM);
+        // Local header extra field length
+        short elenLoc = buffer.getShort(ZipFile.LOCEXT);
 
-    /**
-     * Utility Method for combining the output from a ProcessBuilder invocation
-     * @param in1 ProccessBuilder.getInputStream
-     * @param in2 ProcessBuilder.getErrorStream
-     * @return The ProcessBuilder output
-     * @throws IOException if an error occurs
-     */
-    static String toString(InputStream in1, InputStream in2) throws IOException {
-        try (ByteArrayOutputStream dst = new ByteArrayOutputStream();
-             InputStream concatenated = new SequenceInputStream(in1, in2)) {
-            concatenated.transferTo(dst);
-            return new String(dst.toByteArray(), StandardCharsets.UTF_8);
-        }
-    }
+        // Offset of the CEN header
+        int cenOff = ZipFile.LOCHDR + nlenLoc + elenLoc;
 
-    /**
-     * Utility class used to hold the results from  a ProcessBuilder execution
-     */
-    static class Result {
-        final int ec;
-        final String output;
+        // Read name, extra field and comment lengths
+        short nlen = buffer.getShort(cenOff + ZipFile.CENNAM);
+        short elen = buffer.getShort(cenOff + ZipFile.CENEXT);
 
-        private Result(int ec, String output) {
-            this.ec = ec;
-            this.output = output;
-        }
-        Result assertSuccess() {
-            assertTrue(ec == 0, "Expected ec 0, got: ", ec, " , output [", output, "]");
-            return this;
-        }
-    }
-    static void assertTrue(boolean cond, Object ... failedArgs) {
-        if (cond)
-            return;
-        StringBuilder sb = new StringBuilder();
-        for (Object o : failedArgs)
-            sb.append(o);
-        Assert.fail(sb.toString());
+        // Update CEN sizes and loc offset to 0'xFFFFFFFF
+        buffer.putInt(cenOff + ZipFile.CENLEN, ZIP64_MAGIC_VALUE);
+        buffer.putInt(cenOff + ZipFile.CENSIZ, ZIP64_MAGIC_VALUE);
+        buffer.putInt(cenOff + ZipFile.CENOFF, ZIP64_MAGIC_VALUE);
+
+        // Offset of the extra fields
+        int extraOff = cenOff + ZipFile.CENHDR + nlen;
+
+        // Position at the start of the Zip64 extra field
+        buffer.position(extraOff + elen - ZIP64_SIZE);
+
+        // Update the Zip64 field with the real tag
+        buffer.putShort((short) 0x1); //  Tag for Zip64
```

Comment:
I would probably make 0x1 a constant like the unknown tag.

It would be good to beef up the comment as the byte array written by ZipEntry::setExtra is set to the unknown tag for the Tag and the length is set to 28 which includes all of the Zip64 extra fields

Reply:
> I would probably make 0x1 a constant like the unknown tag.

Extracted a constant with a comment.

> It would be good to beef up the comment as the byte array written by ZipEntry::setExtra is set to the unknown tag for the Tag and the length is set to 28 which includes all of the Zip64 extra fields

Opted to extract a method to create the opaque extra field since that allows for more comments and declutters the calling method.